integration-tests/smoke/ccip/ccip_reorg_test.go

```diff
@@ -0,0 +1,556 @@
+	sourceSelector := allChains[0]
+	destSelector := allChains[1]
+
+	// Chain setup
+	chainSelToRPCURL := buildChainSelectorToRPCURLMap(t, dockerEnv)
+	destClient := ctf_client.NewRPCClient(chainSelToRPCURL[destSelector], nil)
+
+	// Test setup
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, sourceSelector, destSelector, false)
+	waitForLogPollerFilters(l)
+
+	// Initial operation
+	msg := sendCCIPMessage(t, e.Env, state, sourceSelector, destSelector, l)
+	_, err := testhelpers.ConfirmCommitWithExpectedSeqNumRange(
+		t,
+		sourceSelector,
+		e.Env.Chains[destSelector],
+		state.Chains[destSelector].OffRamp,
+		nil, // startBlock
+		ccipocr3.NewSeqNumRange(1, 1),
+		false, // enforceSingleCommit
+	)
+	require.NoError(t, err, "Commit verification failed before reorg")
+
+	// Reorg execution
+	minBlock := msg.Raw.BlockNumber + lessThanFinalityReorgDepth - 1
+	waitForBlockNumber(t, destClient, minBlock, 1*time.Minute, 500*time.Millisecond, l)
+	performReorg(t, destClient, lessThanFinalityReorgDepth, l)
+
+	// Post-reorg validation
+	_, err = testhelpers.ConfirmCommitWithExpectedSeqNumRange(
+		t,
+		sourceSelector,
+		e.Env.Chains[destSelector],
+		state.Chains[destSelector].OffRamp,
+		nil, // startBlock
+		ccipocr3.NewSeqNumRange(1, 1),
+		false, // enforceSingleCommit
+	)
+	require.NoError(t, err, "Commit verification failed after reorg")
+}
+
+func Test_CCIPReorg_GreaterThanFinality_OnDest(t *testing.T) {
+	e, l, dockerEnv, nonBootstrapP2PIDs, state := setupReorgTest(t, logsToIgnoreOpt)
+
+	allChains := e.Env.AllChainSelectors()
+	require.GreaterOrEqual(t, len(allChains), 2)
+	sourceSelector := allChains[0]
+	destSelector := allChains[1]
+	l.
+		Info().
+		Uint64("sourceSelector", sourceSelector).
+		Uint64("destSelector", destSelector).
+		Msg("Chain selectors")
+
+	// Chain setup
+	chainSelToRPCURL := buildChainSelectorToRPCURLMap(t, dockerEnv)
+	destClient := ctf_client.NewRPCClient(chainSelToRPCURL[destSelector], nil)
+
+	// Test setup
+	testhelpers.AddLaneWithDefaultPricesAndFeeQuoterConfig(t, &e, state, sourceSelector, destSelector, false)
+	waitForLogPollerFilters(l)
+
+	// Initial operation
+	sendCCIPMessage(t, e.Env, state, sourceSelector, destSelector, l)
+	_, err := testhelpers.ConfirmCommitWithExpectedSeqNumRange(
+		t,
+		sourceSelector,
+		e.Env.Chains[destSelector],
+		state.Chains[destSelector].OffRamp,
+		nil, // startBlock
+		ccipocr3.NewSeqNumRange(1, 1),
+		false, // enforceSingleCommit
+	)
+	require.NoError(t, err, "Commit verification failed before reorg")
+
+	// Deep reorg execution
+	performReorg(t, destClient, greaterThanFinalityReorgDepth, l)
+	nodeAPIs := dockerEnv.GetCLClusterTestEnv().ClCluster.NodeAPIs()
+
+	// Finality violation check
+	checkFinalityViolations(
+		t,
+		nodeAPIs,
+		nonBootstrapP2PIDs,
+		getHeadTrackerService(t, destSelector),
+		getLogPollerService(t, destSelector),
+		l,
+		len(nodeAPIs)-1,
+		tests.WaitTimeout(t),
+		10*time.Second,
+	)
+
+	// Commit absence check
```

Comment:
Should we skip the test if this check doesn't work?

Reply:
I know it might be a bit brittle, but maybe we should keep polling for, let's say 1 minute to check for commit report absence like you did in the commented code. We can assume that in 99% cases if CCIP is not halted, Commit will be there in 1 minute at most. Otherwise, it means that processing stopped and test is green, because nothing was published.